components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImpl.java

```diff
@@ -221,6 +231,137 @@ public void deleteBrandingPreference(String type, String name, String locale)
         }
     }
 
+    @Override
+    public CustomText addCustomText(CustomText customText) throws BrandingPreferenceMgtException {
+
+        String resourceName = getResourceNameForCustomText(customText.getScreen(), customText.getLocale());
+        String tenantDomain = getTenantDomain();
+        // Check whether a custom text resource already exists with the same name in the particular tenant to be added.
+        if (isResourceExists(CUSTOM_TEXT_RESOURCE_TYPE, resourceName)) {
+            throw handleClientException(ERROR_CODE_CUSTOM_TEXT_ALREADY_EXISTS, tenantDomain);
+        }
+        String preferencesJSON = generatePreferencesJSONFromPreference(customText.getPreference());
+        if (!BrandingPreferenceMgtUtils.isValidJSONString(preferencesJSON)) {
+            throw handleClientException(ERROR_CODE_INVALID_CUSTOM_TEXT_PREFERENCE, tenantDomain);
+        }
+
+        try {
+            InputStream inputStream = new ByteArrayInputStream(preferencesJSON.getBytes(StandardCharsets.UTF_8));
+            Resource customTextPreferenceResource = buildResource(resourceName, inputStream);
+            getConfigurationManager().addResource(CUSTOM_TEXT_RESOURCE_TYPE, customTextPreferenceResource);
+        } catch (ConfigurationManagementException e) {
+            if (RESOURCE_ALREADY_EXISTS_ERROR_CODE.equals(e.getErrorCode())) {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Custom Text preferences are already exists for tenant: " + tenantDomain, e);
+                }
+                throw handleClientException(ERROR_CODE_CUSTOM_TEXT_ALREADY_EXISTS, tenantDomain);
+            }
+            throw handleServerException(ERROR_CODE_ERROR_ADDING_CUSTOM_TEXT_PREFERENCE, tenantDomain, e);
+        }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Custom Text preference for tenant: " + tenantDomain + " added successfully");
+        }
+        return customText;
+    }
+
+    @Override
+    public CustomText getCustomText(String type, String name, String screen, String locale)
+            throws BrandingPreferenceMgtException {
+
+        String resourceName = getResourceNameForCustomText(screen, locale);
+        String tenantDomain = getTenantDomain();
+        try {
+            List<ResourceFile> resourceFiles =
+                    getConfigurationManager().getFiles(CUSTOM_TEXT_RESOURCE_TYPE, resourceName);
+            if (resourceFiles.isEmpty()) {
+                LOG.info("resourceFiles.isEmpty()");
+                throw handleClientException(ERROR_CODE_CUSTOM_TEXT_PREFERENCE_NOT_EXISTS, tenantDomain);
+            }
+            if (StringUtils.isBlank(resourceFiles.get(0).getId())) {
+                LOG.info("StringUtils.isBlank(resourceFiles.get(0).getId())");
```

Comment:
Remove the unwanted info logs

Reply:
Adddressed with https://github.com/wso2-extensions/identity-branding-preference-management/commit/fdd3486cfdc51558c723795cf2fb75b6faafcdc5